src/main/java/cn/edu/thu/tsfile/timeseries/write/series/SeriesWriterImpl.java

```diff
@@ -1,224 +1,281 @@
+		}
+	}
+
+	private void resetPageStatistics() {
+		this.pageStatistics = Statistics.getStatsByType(dataType);
+	}
+
+	@Override
+	public void write(long time, long value) throws IOException {
+		this.time = time;
+		++valueCount;
+		dataValueWriter.write(time, value);
+		pageStatistics.updateStats(value);
+		if (cacheCurrentPageData != null) {
+			cacheCurrentPageData.putTime(time);
+			cacheCurrentPageData.putLong(value);
+		}
+		if (minTimestamp == -1)
+			minTimestamp = time;
+		checkPageSize();
+	}
+
+	@Override
+	public void write(long time, int value) throws IOException {
+		this.time = time;
+		++valueCount;
+		dataValueWriter.write(time, value);
+		pageStatistics.updateStats(value);
+		if (cacheCurrentPageData != null) {
+			cacheCurrentPageData.putTime(time);
+			cacheCurrentPageData.putInt(value);
+		}
+		if (minTimestamp == -1)
+			minTimestamp = time;
+		checkPageSize();
+	}
+
+	@Override
+	public void write(long time, boolean value) throws IOException {
+		this.time = time;
+		++valueCount;
+		dataValueWriter.write(time, value);
+		pageStatistics.updateStats(value);
+		if (cacheCurrentPageData != null) {
+			cacheCurrentPageData.putTime(time);
+			cacheCurrentPageData.putBoolean(value);
+		}
+		if (minTimestamp == -1)
+			minTimestamp = time;
+		checkPageSize();
+	}
+
+	@Override
+	public void write(long time, float value) throws IOException {
+		this.time = time;
+		++valueCount;
+		dataValueWriter.write(time, value);
+		pageStatistics.updateStats(value);
+		if (cacheCurrentPageData != null) {
+			cacheCurrentPageData.putTime(time);
+			cacheCurrentPageData.putFloat(value);
+		}
+		if (minTimestamp == -1)
+			minTimestamp = time;
+		checkPageSize();
+	}
+
+	@Override
+	public void write(long time, double value) throws IOException {
+		this.time = time;
+		++valueCount;
+		dataValueWriter.write(time, value);
+		pageStatistics.updateStats(value);
+		if (cacheCurrentPageData != null) {
+			cacheCurrentPageData.putTime(time);
+			cacheCurrentPageData.putDouble(value);
+		}
+		if (minTimestamp == -1)
+			minTimestamp = time;
+		checkPageSize();
+	}
+
+	@Override
+	public void write(long time, BigDecimal value) throws IOException {
+		this.time = time;
+		++valueCount;
+		dataValueWriter.write(time, value);
+		pageStatistics.updateStats(value);
+		if (minTimestamp == -1)
+			minTimestamp = time;
+		checkPageSize();
+	}
+
+	@Override
+	public void write(long time, Binary value) throws IOException {
+		this.time = time;
+		++valueCount;
+		dataValueWriter.write(time, value);
+		pageStatistics.updateStats(value);
+		if (cacheCurrentPageData != null) {
+			cacheCurrentPageData.putTime(time);
+			cacheCurrentPageData.putBinary(value);
+		}
+		if (minTimestamp == -1)
+			minTimestamp = time;
+		checkPageSize();
+	}
+
+	@Override
+	public List<Object> query() {
+
+		Pair<List<ByteArrayInputStream>, CompressionTypeName> pagePairData = pageWriter.query();
+		DynamicOneColumnData ret = null;
```

Comment:
this initializer 'null' is redundant, merge it and the next line into a single line